Shared/sdk/SharedUtil.Crypto.h

```diff
@@ -28,4 +31,48 @@ namespace SharedUtil
 
         return result;
     }
+
+    inline std::pair<SString, SString> Aes128encode(const SString& sData, const SString& sKey)
+    {
+        using namespace CryptoPP;
+
+        AutoSeededRandomPool rnd;
+
+        SString result;
+        SString sIv;
+
+        SecByteBlock key(0x00, AES::DEFAULT_KEYLENGTH);
+        SecByteBlock iv(0x00, AES::BLOCKSIZE);
+
+        memcpy(key, sKey.c_str(), sKey.size());
+
+        sIv.resize(sizeof(iv));
+        rnd.GenerateBlock((CryptoPP::byte*)sIv.data(), sIv.size());
+        memcpy(iv, sIv.c_str(), sIv.size());
+
+        CTR_Mode<AES>::Encryption aesEncryption;
+        aesEncryption.SetKeyWithIV(key, sizeof(key), iv);
+        StringSource ss(sData, true, new StreamTransformationFilter(aesEncryption, new StringSink(result)));
+
+        return std::pair<SString, SString>(result, sIv);
+    }
+
+    inline SString Aes128decode(const SString& sData, const SString& sKey, const SString& sIv)
+    {
+        using namespace CryptoPP;
+
+        SString result;
+
+        SecByteBlock key(0x00, AES::DEFAULT_KEYLENGTH);
+        SecByteBlock iv(0x00, AES::BLOCKSIZE);
+
+        memcpy(key, sKey.c_str(), sKey.size());
+        memcpy(iv, sIv.c_str(), sIv.size());
+
+        CTR_Mode<AES>::Decryption aesDecryption;
+        aesDecryption.SetKeyWithIV(key, sizeof(key), iv);
```

Comment:
Use this, and delete the `SecByteBlock` stuff.
```cpp
        aesDecryption.SetKeyWithIV(sKey.c_str(), sKey.length(), sIv.c_str());
```
Also, do some checks before calling this function, as to be `sIv.length() >= AES::BLOCKSIZE`

Reply:
Fixed